code/edge/src/test/java/com/adobe/marketing/mobile/EdgeHitProcessorTests.java

```diff
@@ -1287,6 +1217,256 @@ public void setLocationHint(final String hint, final int ttlSeconds) {
 		assertFalse(payloadCaptor.getValue().contains("implementationdetails"));
 	}
 
+	@Test
+	public void testProcessHit_experienceEvent_withDatastreamIdOverrideSet_sendsNetworkRequest_returnsTrue() {
+		// Tests that when a good hit is processed that a network request is made and the request returns 200
+
+		// setup
+		EdgeDataEntity entity = new EdgeDataEntity(
+			getExperienceEventWithConfig("testDatastreamId", null),
+			edgeConfig,
+			identityMap
+		);
+
+		// test
+		assertProcessHit(entity, true, true);
+	}
+
+	@Test
+	public void testProcessHit_experienceEvent_withDatastreamConfigOverrideSet_sendsNetworkRequest_returnsTrue() {
+		// Tests that when a good hit is processed that a network request is made and the request returns 200
+		Map<String, Object> configOverrides = new HashMap<>();
+		configOverrides.put("key", "value");
+
+		// setup
+		EdgeDataEntity entity = new EdgeDataEntity(
+			getExperienceEventWithConfig(null, configOverrides),
+			edgeConfig,
+			identityMap
+		);
+
+		// test
+		assertProcessHit(entity, true, true);
+	}
+
+	@Test
+	public void testProcessHit_experienceEvent_withDatastreamIdAndConfigOverrideSet_sendsNetworkRequest_returnsTrue() {
+		// Tests that when a good hit is processed that a network request is made and the request returns 200
+		Map<String, Object> configOverrides = new HashMap<>();
+		configOverrides.put("key", "value");
+
+		// setup
+		EdgeDataEntity entity = new EdgeDataEntity(
+			getExperienceEventWithConfig("testDatastreamId", configOverrides),
+			edgeConfig,
+			identityMap
+		);
+
+		// test
+		assertProcessHit(entity, true, true);
+	}
+
+	@Test
+	public void testProcessHit_experienceEvent_whenDatastreamIdOverrideSet_sendsSdkConfig() throws Exception {
+		mockNetworkServiceResponse("https://test.com", new RetryResult(EdgeNetworkService.Retry.NO));
+
+		Map<String, Object> configOverrides = new HashMap<>();
+		configOverrides.put("key", "value");
+
+		DataEntity dataEntity = new EdgeDataEntity(
+			getExperienceEventWithConfig("testDatastreamId", null),
+			edgeConfig,
+			identityMap
+		)
+			.toDataEntity();
+		// test
+		hitProcessor =
```

Comment:
Why do you need to recreate the hit processor here? Can you add a note stating the differences with the default?